components/molecules/GameCard/GameCard.tsx

```diff
@@ -0,0 +1,221 @@
+          shadowOpacity: 0.8,
+          shadowRadius: 2,
+          elevation: 5,
+        },
+        imageContainer: {
+          padding: theme.spacing.md,
+        },
+        footerContainer: {
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          marginTop: theme.spacing.lg,
+        },
+        image: {
+          height: 120,
+          width: 120,
+          borderRadius: theme.roundness.xs,
+          borderWidth: 0.5,
+          borderColor: theme.colors.neutral + '99',
+        },
+        miniImage: {
+          width: 38,
+          height: 38,
+          borderRadius: theme.roundness.xs,
+          borderWidth: 0.5,
+          marginRight: 8,
+          borderColor: theme.colors.neutral + '99',
+        },
+        miniImageWithSubtitle: {
+          width: 32,
+          height: 32,
+          marginRight: 10,
+          borderRadius: theme.roundness.xs,
+          borderWidth: 0.5,
+          borderColor: theme.colors.neutral + '99',
+        },
+        name: {
+          width: 120,
+          height: 32,
+          color: theme.colors.neutral,
+          marginTop: theme.spacing.md,
+        },
+        notificationLabelContainer: {
+          position: 'absolute',
+          top: 0,
+          right: 0,
+        },
+        miniContainer: {
+          flexDirection: 'row',
+          alignItems: 'center',
+        },
+      }),
+    [theme],
+  )
+
+  let content = <Fragment />
+
+  if (variation === 'full') {
+    content = (
+      <View style={styles.container}>
+        {/* Card Main */}
+        <View style={styles.imageContainer}>
+          <Image
+            source={{ uri: imageUrl }}
+            style={styles.image}
+            resizeMode="cover"
+          />
+          <Title numberOfLines={2} variation="subtitle2" style={styles.name}>
+            {name}
+          </Title>
+        </View>
+
+        {/* Card Footer */}
+        <View style={styles.footerContainer}>
+          <LikeStat likes={likesByFriends!} suffix="friends" />
+          <LikeStat likes={likesOnPlatform!} />
+        </View>
+
+        {notificationsCount > 0 && (
+          <View style={styles.notificationLabelContainer}>
+            <Label count={notificationsCount} />
+          </View>
+        )}
+      </View>
+    )
+  } else if (variation === 'mini') {
+    content = (
+      <View>
+        <Image source={{ uri: imageUrl }} style={styles.image} />
+        <Title numberOfLines={2} variation="subtitle2" style={styles.name}>
+          {name}
+        </Title>
+      </View>
+    )
+  } else if (variation === 'icon') {
+    if (callout) {
+      content = (
+        <View style={styles.miniContainer}>
+          <Image
+            source={{ uri: imageUrl }}
+            style={styles.miniImageWithSubtitle}
+          />
+          <View>
+            <Title variation="subtitle1">{name}</Title>
+            <Text>{callout}</Text>
+          </View>
+        </View>
+      )
+    } else {
```

Comment:
@J-Rehan I'm confused by this default case. The only supported options are full, mini, and icon so what is this default mode that is rendered? Shouldn't it default to either full, mini, or icon?

Reply:
In the Figma design, we have 2 icon versions.
1. Without the callout text
    https://www.figma.com/file/hITB12zwLVyi9BI2zFhmcU/Game-card-component%3A-Rehan-(Copy)?node-id=303%3A448&t=Ac5se3dhkSxpkonf-4
2. With callout text
    https://www.figma.com/file/hITB12zwLVyi9BI2zFhmcU/Game-card-component%3A-Rehan-(Copy)?node-id=303%3A439&t=ESg1q8wCGRssJeM1-4

They have **different image sizes** (38x38 for icon variation without callout text & 32 x 32 for icon variation with callout text) and **different gaps** between the image and text.

So instead of creating a different sub-variation for icon variation, I have `if-else` conditions for without & with callouts.